src/loaders/svg/tvgSvgSceneBuilder.cpp

```diff
@@ -304,14 +312,21 @@ static void _applyComposition(SvgLoaderData& loaderData, Paint* paint, const Svg
 }
 
 
-static void _applyProperty(SvgLoaderData& loaderData, SvgNode* node, Shape* vg, const Box& vBox, const string& svgPath, bool clip)
+static inline void _setFill(Shape* shape, uint8_t r, uint8_t g, uint8_t b, uint8_t a)
 {
-    SvgStyleProperty* style = node->style;
+    shape->fill(r, g, b, a);
+}
 
-    //Clip transformation is applied directly to the path in the _appendClipShape function
-    if (node->transform && !clip) vg->transform(*node->transform);
-    if (node->type == SvgNodeType::Doc || !node->style->display) return;
 
+static inline void _setFill(Text* text, uint8_t r, uint8_t g, uint8_t b, TVG_UNUSED uint8_t a)
+{
+    text->fill(r, g, b);
+}
+
+
+template<typename Vg, typename std::enable_if<std::is_same<Vg, Shape*>::value || std::is_same<Vg, Text*>::value, int>::type = 0>
```

Comment:
@hermet please comment whether we are ok with such template

Reply:
Please reach a consensus with @JSUYA. I will follow you guys consensus. In my opinion, using complicate templates will hinder the ThorVG project, as newcomers will face a significant barrier in understanding the code. I would not prefer it unless it helps to either a) reduce massive duplicate code (without increasing the binary size much) or b) enhance performance.

Reply:
I don't like this solution (I wrote it :D ) because we still have the same code for a stroke.